// react-chevron/Chevron/Chevron.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getChevronStyle = exports.ChevronWrapper = void 0;
exports.ChevronWrapper = {
    marginLeft: '0.2em',
    marginRight: '0.36em',
    overflow: 'visible'
};
const ChevronWrapper__SVG = {
    verticalAlign: 'baseline',
    transition: '.25s ease-in-out',
    fill: 'currentColor'
};
const ChevronWrapper_SVG__UpDownIcon = {
    width: '.8em'
};
const ChevronWrapper_SVG__Up = {
    marginBottom: '.1em'
};
const ChevronWrapper__SVG_RightLeft = {
    height: '.7em',
    marginRight: '.12em',
    marginBottom: '-.08em'
};
const ChevronWrapper__Reversed = {
    transform: 'rotate(180deg)'
};
const getChevronStyle = (direction, reversed) => {
    let styleObject = Object.assign({}, ChevronWrapper__SVG);
    switch (direction) {
        case 'up':
            styleObject = Object.assign(Object.assign(Object.assign({}, styleObject), ChevronWrapper_SVG__UpDownIcon), ChevronWrapper_SVG__Up);
            break;
        case 'down':
            styleObject = Object.assign(Object.assign({}, styleObject), ChevronWrapper_SVG__UpDownIcon);
            break;
        case 'left':
        case 'right':
            styleObject = Object.assign(Object.assign({}, styleObject), ChevronWrapper__SVG_RightLeft);
    }
    if (reversed)
        styleObject = Object.assign(Object.assign({}, styleObject), ChevronWrapper__Reversed);
    return styleObject;
};
exports.getChevronStyle = getChevronStyle;
